sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -356,8 +358,28 @@ case class PreprocessTableInsertion(conf: SQLConf) extends Rule[LogicalPlan] {
           s"including ${staticPartCols.size} partition column(s) having constant value(s).")
     }
 
-    val newQuery = DDLPreprocessingUtils.castAndRenameQueryOutput(
-      insert.query, expectedColumns, conf)
+    val newQuery = if (conf.getConf(SQLConf.LEGACY_INSERT_TABLE_TYPE_COERCION)) {
+      DDLPreprocessingUtils.castAndRenameQueryOutput(insert.query, expectedColumns, conf)
+    } else {
+      val errors = new mutable.ArrayBuffer[String]()
```

Comment:
Why not run `ResolveOutputRelation`?

Reply:
Here is for the V1 path. `InsertIntoTable` won't be matched in `ResolveOutputRelation`.

Reply:
If you aren't going to use the `ResolveOutputRelation` rule, can you explain exactly what the behavior this code path should have and how it differs from `ResolveOutputRelation`?

Reply:
I check and I think it is workable. I will try moving it into the `ResolveOutputRelation` rule.

Reply:
Have we moved it?

Reply:
We need to figure out if we can up-cast decimal to double/float. 
If we can't, then maybe we can't continue this PR.
I have created a new PR https://github.com/apache/spark/pull/24849

Reply:
@gengliangwang, why is decimal to double/float required? That should not be allowed in either v1 or v2, so I see no reason why we can't use the same rule for both.

Reply:
For sql
```
create table t (d double);
insert into t values (10.0);
```
the 10.0 is decimal in Spark SQL parser.